### PR TITLE
slides 0.1.2 (new formula)

### PR DESCRIPTION
**Pull request by maaslalani**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/slides.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args
+  end
+
+  test do
+    assert_match "slides <file.md> [flags]", shell_output("#{bin}/slides")
```
Comment (nandahkrishna) [inline Formula/slides.rb:15]: We need a test that exercises some actual functionality of the program, and not one that just compares `help` or `version` output. Could you propose a better test here?

Comment (maaslalani) [reply Formula/slides.rb:15]: I may need to change the program since currently it needs a `tty`.

Comment (alebcay) [reply Formula/slides.rb:15]: We do have ways to simulate a TTY (a PTY) in Homebrew tests. [Something like how these formulae do it](https://github.com/Homebrew/homebrew-core/search?q=pty).

Comment (maaslalani) [reply Formula/slides.rb]: I found a similar formula that is very similar to my application (Using bubbletea). Looks like they had similar problems with `PTY` as well.
https://github.com/Homebrew/homebrew-core/blob/11d3369762a7eccf2349eefc732ffe183cb0ba13/Formula/tz.rb

I added a similar test to their formula which should test that the program was at least able to parse the input passed into it.

File: Formula/slides.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  head "https://github.com/maaslalani/slides.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args
+  end
+
+  test do
+    markdown = <<-SLIDES
```
Comment (carlocab) [inline Formula/slides.rb:16]: Can we write this to a file instead, and open the file with `slides`? Pipes sometimes produce orphan processes that `brew` isn't able to kill after the test times out, and this makes some CI runs deadlock.

Comment (maaslalani) [reply Formula/slides.rb:16]: 👍

Comment (maaslalani) [reply Formula/slides.rb]: Done 🙂

**Review by dawidd6 — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by nandahkrishna** [comment]
Thanks, @maaslalani!

**Comment by maaslalani** [comment]
Thank you folks! ❤️

**Pull request closed without merging**